features/comments_and_kudos/spam_comments.feature

```diff
@@ -46,3 +46,31 @@ Feature: Marking comments as spam
 
     When I follow "Default Admin Post"
     Then I should see "Comments (1)"
+
+  Scenario: Author can mark comments as spam
+    Given I am logged in as "author"
+      And I post the work "Popular Fic"
+      And I log out
+    When I view the work "Popular Fic" with comments
+      And I post a spam comment
+      And I post a guest comment
+    When I am logged in as "author"
+      And I view the work "Popular Fic" with comments
+      Then I should see "Comments (2)"
+      Then I should see "Buy my product"
+    When I mark the comment as spam
+      Then I should see "Comments (1)"
+      Then I should not see "Buy my product"
+      Then all comments by "spammer" are marked as spam
```

Comment:
It looks like the indenting and `Then`s and `When`s went a little awry here:
```suggestion
      And I am logged in as "author"
      And I view the work "Popular Fic" with comments
    Then I should see "Comments (2)"
      And I should see "Buy my product"
    When I mark the comment as spam
    Then I should see "Comments (1)"
      And I should not see "Buy my product"
```

I took out the `Then all comments by "spammer" are marked as spam` step because that's not actually a `Then` that is checking the results, but rather a `When` that is performing an action (which is unnecessary in this scenario):

https://github.com/otwcode/otwarchive/blob/ff9f334bb93d69f2f1c38d39fe95cf97a0afec57/features/step_definitions/comment_steps.rb#L171-L173

Reply:
> I took out the Then all comments by "spammer" are marked as spam step because that's not actually a Then that is checking the results, but rather a When that is performing an action

Oh yeah, thanks, read too quickly and indeed thought it was a check not an action.